### ISSUE TITLE
[$500] IOU - Inconsistency in currency displayed while making distance request

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#30595]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.93-0
**Reproducible in staging?:**Y
**Reproducible in production?:**Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Launch app
2. Tap profile icon
3. Tap workspaces
4. Tap on workspace
5. Tap on workspace image
6. Change default currency to dollar
7. Tap save
8. Navigate back to LHN
9. Tap on the Workspace chat
10. Tap plus icon--Request money --distance
11. Enter start and finish waypoints
12. Tap next
13. Tap request and note currency $ displayed in amount and distance per rate
14. Navigate back to Workspace page and change default currency to UAH and AWG
15. Navigate back to LHN
16. Tap on the Workspace chat
17. Perform steps 10- step 12
18. Tap request and note UAH and AWG shown in amount and it's currency symbol displayed in distance per rate.

## Expected Result:
While making distance request, the currency symbol of default currency in WS, must be shown in both amount and distance per rate section

## Actual Result:
While making distance request with dollar as default currency in WS, $ symbol is shown in both amount and distance per rate. But when UAH and AWG is used as default currency in Workspace, in amount UAH and AWG is displayed and it's currency symbol is displayed only at distance per rate section. Inconsistency in currency displayed

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
<details>
<summary>Android: Native</summary>

 

https://github.com/Expensify/App/assets/78819774/7d42611e-1f36-4937-9926-a9575a97c996

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~013b6d8002412236ae</li>
        <li>Upwork Job ID: 1719024622344015873</li>
        <li>Last Price Increase: 2023-11-06</li>
    </ul>
</details>

**Comment by rakshitjain13** [comment]
Dupe of https://github.com/Expensify/App/issues/29822

**Comment by rakshitjain13** [comment]
Similar to https://github.com/Expensify/App/issues/29822

**Comment by DylanDylann** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
 Inconsistency in currency displayed while making distance request
### What is the root cause of that problem?
We use the different logic to render currency in 2 places:
1. In Amount field: we use Intl.NumberFormat (used ISO 4217)
https://github.com/Expensify/App/blob/72b388a4021fcf541b8f825f902d9cf34fc1c6f6/src/libs/CurrencyUtils.ts#L108-L115
2. In Distance field: we use getCurrencySymbol function to get currency symbol
https://github.com/Expensify/App/blob/72b388a4021fcf541b8f825f902d9cf34fc1c6f6/src/libs/CurrencyUtils.ts#L57-L59
It causes the inconsistency
### What changes do you think we should make in order to solve the problem?
To make it consistent, we need to decide one of two above ways is suitable and update both way to use a same logic


### What alternative solutions did you explore? (Optional)

**Comment by Kalomidin** [comment]
# Proposal
---
## Please re-state the problem that we are trying to solve in this issue.
Inconsistency in currency displayed while making distance request

## What is the root cause of that problem?
You can check in here:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat

It sometimes uses symbol and sometimes uses currency. However, in the code in here:
https://github.com/Expensify/App/blob/72b388a4021fcf541b8f825f902d9cf34fc1c6f6/src/libs/CurrencyUtils.ts#L57-L59

It uses map to get the symbol instead of using NumberFormat. Solution is to be consistent on usage of currencySymbol.

## What changes do you think we should make in order to solve the problem?
N/A

**Comment by Kalomidin** [comment]
Contributor details
Your Expensify account email: t.kalom1994@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01426381604c4e1791

**Comment by sonialiap** [comment]
Hmm this one does look very similar to https://github.com/Expensify/App/issues/29822. @Santhosh-Sellavel do you think this one may be a dupe?

**Comment by sonialiap** [comment]
To me it looks like a dupe, but waiting on confirmation from Santhosh or Robert (in the other issue)

**Comment by Santhosh-Sellavel** [comment]
Sorry for the delay it's not a duplicate.

**Comment by cead22** [comment]
I disagree. I think we should close this issue or put it on hold. More context here https://github.com/Expensify/App/issues/29822#issuecomment-1804315422

**Comment by sonialiap** [comment]
Closing out per the discussion in https://github.com/Expensify/App/issues/29822
